packages/protocol-contracts/test/deterministicDeployment.index.utils.ts

```diff
@@ -0,0 +1,54 @@
+import { Provider, TransactionReceipt } from "@ethersproject/providers";
+import assert from "assert";
+import { ethers, Signer } from "ethers";
+
+import { ImmutableCreate2Factory__factory } from "../typechain-types";
+import { buildBytecode, buildCreate2Address, saltToHex } from "./deterministicDeployment.utils";
+
+export async function deployContract({
```

Comment:
Maybe we can rename to deployDeterministicAddressContract? A bit long but useful so we don't confuse it with a generic deploy contract function.